docs/html/reference/pip_install.rst

```diff
@@ -871,6 +871,11 @@ Examples
 
       $ pip install --upgrade SomePackage
 
+    .. note::
+
+      This will only update ``SomePackage`` as it is a direct requirement. Any
+      of its dependencies (indirect requirements) will be affected by the
+      ``--upgrade-strategy`` command.
```

Comment:
Indirect requirements can still be upgraded here if a later version of `SomePackage` needs them upgraded. Say

* `SomePackage==1.0` needs `DepPackage==1.0`
* `SomePackage==2.0` needs `DepPackage==2.0`

With `SomePackage==1.0` and `DepPackage==1.0` initially installed, `pip install --upgrade SomePackage` will still upgrade `DepPackage`.

Reply:
IOW, only `SomePackage` is *guarenteed* to be upgraded. Its dependencies are upgraded only if needed.